numpy/lib/npyio.pyi

```diff
@@ -50,7 +53,17 @@ _DTypeLike = Union[
 ]
 
 class _SupportsGetItem(Protocol[_T_contra, _T_co]):
-    def __getitem__(self, key: _T_contra) -> _T_co: ...
+    def __getitem__(self, key: _T_contra, /) -> _T_co: ...
+
+class _SupportsRead(Protocol[_CharType_co]):
+    def read(self) -> _CharType_co: ...
+
+class _SupportsReadSeek(Protocol[_CharType_co]):
+    def read(self, n: int, /) -> _CharType_co: ...
+    def seek(self, offset: int, whence: int, /) -> object: ...
```

Comment:
`np.load` always supplies the full array of optional arguments to `read` and `seek`, so they can be marked as non-optional in the protocol.

Reply:
The name is based of its counterpart in [`io`](https://docs.python.org/3/library/io.html#io.IOBase.seek) and [`typing.IO`](https://github.com/python/cpython/blob/e09dd8aafd6f9aef03945c417267806d47084a5d/Lib/typing.py#L2119-L2121).